### ISSUE TITLE
Evaluation error Missing keys

### ISSUE THREAD

**Issue by programmeddeath1** [HumaticsLAB/SEAM-Match-RCNN#3]
Hi,
 When I am running the evaluation on the pretrained model it is failing with 
'''
RuntimeError: Error(s) in loading state_dict for VideoMatchRCNN:
	**Missing key(s) in state_dict: "backbone.body.conv1.weight", "backbone.body.bn1.weight", "backbone.body.bn1.bias",** "backbone.body.bn1.running_mean", "backbone.body.bn1.running_var", "backbone.body.layer1.0.conv1.weight", "backbone.body.layer1.0.bn1.weight", "backbone.body.layer1.0.bn1.bias", "backbone.body.layer1.0.bn1.running_mean", "backbone.body.layer1.0.bn1.running_var", "backbone.body.layer1.0.conv2.weight", "backbone.body.layer1.0.bn2.weight", "backbone.body.layer1.0.bn2.bias", "backbone.body.layer1.0.bn2.running_mean", "backbone.body.layer1.0.bn2.running_var", "backbone.body.layer1.0.conv3.weight", "backbone.body.layer1.0.bn3.weight", "backbone.body.layer1.0.bn3.bias", "backbone.body.layer1.0.bn3.running_mean", "backbone.body.layer1.0.bn3.running_var", "backbone.body.layer1.0.downsample.0.weight", "backbone.body.layer1.0.downsample.1.weight", roi_heads.temporal_aggregator.newnlb.phi.weight", "roi_heads.temporal_aggregator.newnlb.phi.bias", "roi_heads.temporal_aggregator.newnlb.concat_project.0.weight". 
**Unexpected key(s) in state_dict: "module.backbone.body.conv1.weight", "module.backbone.body.bn1.weight",** "module.backbone.body.bn1.bias", "module.backbone.body.bn1.running_mean", "module.backbone.body.bn1.running_var", "module.backbone.body.layer1.0.conv1.weight", "module.backbone.body.layer1.0.bn1.weight", "module.backbone.body.layer1.0.bn1.bias", "module.backbone.body.layer1.0.bn1.running_mean", "module.roi_heads.match_predictor.conv_seq.4.weight", "module.roi_heads.match_predictor.conv_seq.4.bias", "module.roi_heads.match_predictor.conv_seq.6.weight", "module.roi_heads.match_predictor.conv_seq.6.bias", "module.roi_heads.match_predictor.linear.0.weight", "module.roi_heads.match_predictor.linear.0.bias", "module.roi_heads.match_predictor.linear.1.weight", "module.roi_heads.match_predictor.linear.1.bias", "module.roi_heads.match_predictor.linear.1.running_mean", "module.roi_heads.match_predictor.linear.1.running_var", "module.roi_heads.match_predictor.linear.1.num_batches_tracked", "module.roi_heads.match_predictor.last.weight", "module.roi_heads.match_predictor.last.bias".
'''

Basically all the keys are have module added to the name compared to the config i believe.

May i know if this is an issue with the pretrained model uploaded? if so can you share the corrected one.

Thanks and Regards

**Comment by joppichristian** [comment]
Hi,
when you train the model with DataParallel activated, the saved checkpoint has an extra prefix ("module") at the begging of each key. 
You solve the issue in one of these way:

- training on a single gpu
- saving with model.module.state_dict() instead of model.state_dict()
- activating DataParallel in evaluation, before loading the checkpoint

More about:
https://discuss.pytorch.org/t/missing-keys-unexpected-keys-in-state-dict-when-loading-self-trained-model/22379/3
